### DISCUSSION TITLE
谁有比较好的工作流介绍一下？

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [nshen/learn-neovim-lua#48] [General]
作者能讲一下平时进行前端或者rust开发的工作流么？ 

我自己将配置慢慢改到原本在VSCodoe下的操作，遇到很多坑。

实在VSCode下有一套比较习惯的工作流，感觉应该是可以搬过来的。
但是我想要的某些功能完全找不到应该如何设置快捷键。
比如 ctrl + h 跳转到上一个单词的开头，ctrl + shift +h 跳转到行首开头，完全不知道如何配置。

---
从网上了解到，自己实际上是可以写一些简单的脚本，让vim更加适合自己，这应该如何开始？

---
环境的迁移。按照我的理解，nvim的环境配置还是相当麻烦的，远不如VSCode那么简单。
- 插件查找 插件可靠性无法保证
- Mac Win 两种环境 光是代理问题就是一个大坑，win下的插件安装，我到现在还没有解决。
- Mac Win 两种环境切换 比较难以保证工作流的统一

---

问题多多，现在还无法将vim用到自己的实际开发中去

**Comment by nshen** [comment]
我觉得换到 vim 主要是习惯在 `normal` 模式下做移动， 比如跳到上个单词开头是 `b` ,下一个单词开头是 `w`，跳到行首是 `^` 。

VIM的工作流还是推荐 [Vim实用技巧](https://book.douban.com/subject/26967597/) 那本书哈。

neovim 快捷键设置最新的 API 是 `vim.keymap.set(mode, lhs, rhs, opts)` 可以参考 keybindings.lua 设置尝试修改快捷键。

VSCode 肯定是比 Neovim 强大，毕竟是微软的产品，Neovim 是开源驱动的，还是很原始，非常折腾，只是勉强能用哈。

你说的环境的迁移，我也是渐进的尝试切换到实际开发中的，我经常做着做着项目，发现哪里不好用，然后去搜索，修改 neovim 的配置，折腾好久后再回来继续开发，慢慢磨合让他一点一点变得更好用， 渐渐成为我操作习惯的一部分，用得多了也就习惯了，现在回到 vscode 渐渐会觉得别扭。 

对我来说这是 Neovim 最大的优势是没有 Button 的概念，所有组件都可以键盘控制。VSCode 里就很麻烦，比如 `code action` 还要去用鼠标点那个小按钮，然后下拉列表里再选择一次。Neovim 里 `leader ca` 就可以了，不用离开键盘。

其他的部分功能可能没有 VSCode 的简单好用，但基本上也是可用的。仔细想想写代码就是输入文本，根本不需要那么多花样，太依赖工具也不好，有时候我觉得 Neovim里安装的插件都有点过多了，努力返璞归真中。
